### PR TITLE
Update PEtab tutorial

### PR DESCRIPTION
**Pull request by sebapersson**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API
  
## Additional context

I have updated the code in the PEtab tutorial and added it again to the documentation. The tutorial runs locally for me, but I have problem to precompile the doc environment due to BifurcationKit (which is why the doc runs might fail).

### CODE REVIEW

**Comment by sebapersson** [comment]
I have updated the tutorial to address the minor bugs. The current failure seems unrelated to PEtab, but I do not know why the doc build fails.

**Comment by isaacsas** [comment]
xvfb just randomly fails on GitHub Actions for some reason, but we need it for GLMakie.

**Comment by isaacsas** [comment]
This looks like a real failure in the tutorial.

**Comment by sebapersson** [comment]
The docs should work now :)

File: docs/src/inverse_problems/petab_ode_param_fitting.md [hunk 1]
```diff
@@ -300,12 +295,12 @@ measurements = vcat(m1,m2)
...
+Sometimes, the parameters that are used vary between the different conditions. Consider our catalysis example, if we had performed the experiment twice, using two different enzymes with different catalytic properties, this could have generated such conditions. The two enzymes could e.g. yield different rates ($kP_1$ and $kP_2$) for the `SE --> P + E` reaction, but otherwise be identical. Here, the parameters $kP_1$ and $kP_2$ are unique to their respective conditions. PEtab.jl provides support for cases such as this, and [its documentation](https://sebapersson.github.io/PEtab.jl/stable/petab_cond_specific/) provided instructions of how to handle them.
 
 ## [Additional features: Initial conditions](@id petab_simulation_initial_conditions)
 
 ### [Fitting initial conditions](@id petab_simulation_initial_conditions_fitted)
-Sometimes, initial conditions are uncertain quantities which we wish to fit to the data. This is possible by [defining an initial condition as a parameter](@ref dsl_description_parametric_initial_conditions):
```
Comment (TorkelE) [inline docs/src/inverse_problems/petab_ode_param_fitting.md:308]: any reason to remove this? Seems like a relevant reference?

Comment (sebapersson) [reply docs/src/inverse_problems/petab_ode_param_fitting.md:308]: It failed the link check when building before (thought it had been removed), but that might actually been due to something else (as I can actually find the `@id` block in the code). I agree, I will add it back.

Comment (TorkelE) [reply docs/src/inverse_problems/petab_ode_param_fitting.md:308]: Seems like an excellent reason to remove it. Yes, the link should be `dsl_advanced_options_parametric_initial_conditions`. Probably got changed while PEtab doc was disabled so this wasn't caught somehow.

Comment (TorkelE) [reply docs/src/inverse_problems/petab_ode_param_fitting.md:308]: Ok, seems you found the new link as well

**Comment by isaacsas** [comment]
@sebapersson thanks for the updates!

**Pull request merged**